operator/pkg/awsprovider/instances/reconciler.go

```diff
@@ -123,10 +123,10 @@ func (c *Controller) updateAutoScalingGroup(ctx context.Context, dataplane *v1al
 func (c *Controller) createAutoScalingGroup(ctx context.Context, dataplane *v1alpha1.DataPlane) error {
 	subnets, err := c.subnetsFor(ctx, dataplane)
 	if err != nil {
-		return fmt.Errorf("getting private subnet for %s, %w", dataplane.Spec.ClusterName, err)
+		return fmt.Errorf("getting subnet for %s, %w", dataplane.Spec.ClusterName, err)
 	}
 	if len(subnets) == 0 {
-		return fmt.Errorf("failed to find private subnets for dataplane")
+		return fmt.Errorf("failed to find subnets for dataplane %s", dataplane.Name)
```

Comment:
It would be great if the logger included context such that you didn't need to annotate `dataplane %s", dataplane.Name)`. For example, if at the top level, your logger did something like .With("dataplane", dataplane.Name"), this would magically wire up for all log lines. We do this broadly across karpenter. 

https://github.com/aws/karpenter/blob/main/pkg/controllers/provisioning/provisioner.go#L135

Reply:
yeah, I like the idea. 
This might be a bigger change, not blocking this PR for it.